src/core/thread/mle_router.cpp

```diff
@@ -111,7 +111,24 @@ void MleRouter::HandlePartitionChange(void)
 
 bool MleRouter::IsRouterEligible(void) const
 {
-    return mRouterEligible && IsFullThreadDevice();
+    bool                 rval      = true;
+    const SecurityPolicy secPolicy = Get<KeyManager>().GetSecurityPolicy();
+
+    VerifyOrExit(mRouterEligible && IsFullThreadDevice(), rval = false);
+
+#if OPENTHREAD_CONFIG_THREAD_VERSION == OT_THREAD_VERSION_1_1
+    VerifyOrExit(secPolicy.mRoutersEnabled, rval = false);
+#else
+    if (secPolicy.mCommercialCommissioningEnabled)
+    {
+        VerifyOrExit(secPolicy.mNonCcmRoutersEnabled, rval = false);
+    }
+    VerifyOrExit(secPolicy.mVersionThresholdForRouting + kVersionThresholdOffsetVersion <= kThreadVersion,
```

Comment:
If `R` bit is 1,  Protocol version 1 or 2, or higher could become router, no need to check `VR` bits. That is,  only need check `VR` bits when `R` is 0.

Thoughts?

Reply:
You are right. Guard added.